server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -233,7 +233,10 @@ public static int geoHashLevelsForPrecision(String distance) {
      * @return The normalized longitude.
      */
     public static double normalizeLon(double lon) {
-        return centeredModulus(lon, 360);
+        if (lon > 180d || lon <= -180d) {
+            return centeredModulus(lon, 360);
+        }
+        return lon + 0d;
```

Comment:
can you add a comment that this aims at avoiding negative zeros? Also it's not obvious to me that centerModulus may never return negative zero should we do the same? Also we do we want to avoid negative zeros?

Reply:
I think it would be good to avoid negative 0. We can look at adding that to the parser in a separate PR?  For now I'll add the comment and cleanup so that if centerModulus does return -0.0 its handled appropriately